### ISSUE TITLE
Not compatible with pixi v5. webpack - ReferenceError: PIXI is not defined - MSDFText.ts:338

### ISSUE THREAD

**Issue by milahu** [soimy/pixi-msdf-text#2]
TLDR: please migrate this plugin to pixi v5

i [just found](https://github.com/pixijs/pixi.js/issues/3115#issuecomment-361154206) that this plugin is for pixi v4

so the "PIXI is not defined" error is solved
by adding to `MSDFText.ts`
```js
import * as PIXI from 'pixi.js'
```
and adding a webpack plugin in `webpack.config.js`
```js
module.exports = {
    // ....
    plugins: [
     new webpack.ProvidePlugin({
       PIXI: 'pixi.js'
     })
   ]
}
```
source: [pixi v5 migration guide](https://github.com/pixijs/pixi.js/wiki/v5-Migration-Guide#webpack)

--

im failing at the first mile .. i cant include pixi-msdf-text into my pixi project

what i did:

```sh
#!/bin/bash

# pixi-seed
git clone --depth=1 https://github.com/edwinwebb/pixi-seed.git my-project
cd my-project
rm -rf .git
npm install

# pixi-msdf-text
git clone --depth=1 https://github.com/soimy/pixi-msdf-text.git
cd pixi-msdf-text
npm install
npm run build
cd ..
npm install pixi-msdf-text/
```

add to `app/entry.js`
```js
import MSDFFont from './displayobjects/Text/DimboR.fnt';
// ....
loader.start([BG, LOGO, MSDFFont]);
```
add `*.fnt` file type to `webpack.config.js`
```
      // https://github.com/pixijs/pixi.js/issues/5199
      {
        test: /\.fnt$/,
        use: [
          'file-loader',
          'extract-loader',
          {
            loader: 'html-loader',
            options: {
              // by adding this it will extract each "page” file from the xml as well
              attrs: ['page:file'],
            },
          },
        ],
      },
```

run `npm install -D extract-loader html-loader`

run `npm run start`

the init code seems to work, lets move on

add to `app/screens/Example.js`
```js
import * as MSDFText from 'pixi-msdf-text';
// ....
export default class Example extends Container {
  constructor(...args) {
    // ....
    
    this.addText();
  }
  // ....

  addText(){
    const textOptions = {
        fontFace: "DimboR",
        // ....
    }
    sampleText = new MSDFText.MSDFText("PixiJS support\nMSDF BitmapFont", textOptions);
    this.addChild(sampleText);
  }
```

this fails with "Invalid fontFace" in `MSDFText.ts`
```js
    public updateText() {
        // clear all gizmo
        this.removeChildren();

        const fontData = PIXI.extras.BitmapText.fonts[this._font.fontFace];
        if (!fontData) throw new Error("Invalid fontFace: " + this._font.fontFace);
```
`PIXI.extras.BitmapText.fonts` is empty
either cos its deprecated in pixi v5
or cos the font is not loaded, and the loader fails silently

**Comment by milahu** [comment]
revoke: im using pixi v4.8.8 now, problem solved

**Comment by soimy** [comment]
Sorry, Pixi v5 still need a lot of work(Might need total rewrite), but I don't have that time yet.

**Comment by stevencowles** [comment]
@Soimy -- If it helps -- I have a mostly working Pixi V5 version up here: https://stackblitz.com/edit/steve-wip-pixi-msdf-text-v5 — but there are a few bits I'm not too sure about, which I've listed in the readme. (I've never used Pixi V4).

Feel free to take anything in there that's useful.
